src/memgraph.cpp

```diff
@@ -437,34 +452,62 @@ int main(int argc, char **argv) {
 
   auto system = memgraph::system::System{db_config.durability.storage_directory, FLAGS_data_recovery_on_startup};
 
+  auto const coordination_setup = memgraph::flags::GetFinalCoordinationSetup();
   // singleton replication state
-  memgraph::replication::ReplicationState repl_state{ReplicationStateRootPath(db_config)};
+  memgraph::replication::ReplicationState repl_state{
+      ReplicationStateRootPath(db_config, coordination_setup.management_port != 0),
+      coordination_setup.management_port != 0};
```

Comment:
this sending coordination_setup.mgmt_port != 0 at multiple places doesn't look best

Reply:
Agree, resolved now